pkg/yurthub/cachemanager/handler_layer.go

```diff
@@ -0,0 +1,117 @@
+package cachemanager
+
+import (
+	"sync"
+
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/storage"
+	"k8s.io/klog"
+)
+
+type Handler interface {
+	ServeObject(runtime.Object) (runtime.Object, error)
+}
+
+type HandlerLayer struct {
+	sync.Mutex
+	handlers  map[schema.GroupVersionKind][]Handler
+	selectors map[schema.GroupVersionKind]*storage.SelectionPredicate
```

Comment:
Use `SelectionPredicate` is a good idea, but I think that we need to filter object not only based on labels and fields. How about define a common filter function? like `Filter(obj runtime.Object) bool`

Reply:
The storage.AttrFunc of SelectAndProcess is used to extract the corresponding field data from the actual Object, which is defined by the user. It can basically meet the matching of all fields of runtime.Object. The newly added [hanler_layer_test.go](https://github.com/qclc/openyurt/blob/ec500a29cf37ac70e08b44b2677760d7404ab56c/pkg/yurthub/cachemanager/handler_layer_test.go#L252) has concrete examples to refer to